File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/domain/sample/AbstractCategory.java
package com.coupang.springframework.data.requery.domain.sample;

import com.coupang.springframework.data.requery.domain.AbstractPersistable;
import io.requery.*;

/**
 * AbstractCategory
 *
 * @author debop@coupang.com
 * @since 18. 6. 25
 */
@Entity
public abstract class AbstractCategory extends AbstractPersistable<Long> {

    @Key
    @Generated
    protected Long id;

    @ForeignKey
    @ManyToOne
    protected AbstractProduct product;

    public AbstractCategory() {}

    public AbstractCategory(AbstractProduct product) {
        this.product = product;
    }


    private static final long serialVersionUID = 6433169486936020806L;
}

File: src/test/kotlin/com/coupang/springframework/data/requery/domain/sample/AbstractCategory.kt
package com.coupang.springframework.data.requery.domain.sample

import com.coupang.springframework.data.requery.domain.AbstractPersistable
import io.requery.Entity
import io.requery.Generated
import io.requery.Key
import io.requery.ManyToOne

/**
 * com.coupang.springframework.data.requery.domain.sample.AbstractCategory
 * @author debop
 * @since 18. 5. 30
 */
@Entity
abstract class AbstractCategory(): AbstractPersistable<Long>() {

    protected constructor(product: AbstractProduct): this() {
        this.product = product
    }

    @get:Key
    @get:Generated
    abstract override val id: Long?

    @get:ManyToOne
    abstract var product: AbstractProduct?
}